subprojects/core/src/main/java/org/gradle/initialization/DefaultSettingsLoader.java

```diff
@@ -152,7 +152,14 @@ private void validate(SettingsInternal settings) {
                 String suffix = buildPath == Path.ROOT ? "" : " (in build " + buildPath + ")";
                 throw new GradleException("'" + SettingsInternal.BUILD_SRC + "' cannot be used as a project name as it is a reserved name" + suffix);
             }
+            if (!project.getProjectDir().exists() || !project.getProjectDir().isDirectory() || !project.getProjectDir().canWrite()) {
+                String msgTemplate = "Configuring project '%s' without an existing directory is deprecated. The configured projectDirectory '%s' does not exist, can't be written to or is not a directory.";
```

Comment:
💅 Consider extracting the body block into a static method or `msgTemplate` in a static field.
💅 Consider renaming `msgTemplate` to `warningMessage` or something without shrtng of lttrs.